src/rust/ide/enso-protocol/src/file_manager.rs

```diff
@@ -0,0 +1,411 @@
+    #[MethodInput=CopyFileInput,rpc_name="file/copy",result=copy_file_result,set_result=set_copy_file_result]
+    fn copy_file(&self, from:Path, to:Path) -> ();
+
+    /// Deletes the specified file.
+    #[MethodInput=DeleteFileInput,rpc_name="file/delete",result=delete_file_result,
+    set_result=set_delete_file_result]
+    fn delete_file(&self, path:Path) -> ();
+
+    /// Check if file exists.
+    #[MethodInput=ExistsInput,rpc_name="file/exists",result=exists_result,
+    set_result=set_exists_result]
+    fn exists(&self, path:Path) -> bool;
+
+    /// List all file-system objects in the specified path.
+    #[MethodInput=ListInput,rpc_name="file/list",result=list_result,set_result=set_list_result]
+    fn list(&self, path:Path) -> Vec<Path>;
+
+    /// Moves directory to another location.
+    #[MethodInput=MoveDirectoryInput,rpc_name="file/move",result=move_directory_result,
+    set_result=set_move_directory_result]
+    fn move_directory(&self, from:Path, to:Path) -> ();
+
+    /// Moves file to another location.
+    #[MethodInput=MoveFileInput,rpc_name="file/move",result=move_file_result,
+    set_result=set_move_file_result]
+    fn move_file(&self, from:Path, to:Path) -> ();
+
+    /// Reads file's content as a String.
+    #[MethodInput=ReadInput,rpc_name="file/read",result=read_result,set_result=set_read_result]
+    fn read(&self, path:Path) -> String;
+
+    /// Gets file's status.
+    #[MethodInput=StatusInput,rpc_name="file/status",result=status_result,set_result=set_status_result]
+    fn status(&self, path:Path) -> Attributes;
+
+    /// Creates a file in the specified path.
+    #[MethodInput=TouchInput,rpc_name="file/touch",result=touch_result,set_result=set_touch_result]
+    fn touch(&self, path:Path) -> ();
+
+    /// Writes String contents to a file in the specified path.
+    #[MethodInput=WriteInput,rpc_name="file/write",result=write_result,set_result=set_write_result]
+    fn write(&self, path:Path, contents:String) -> ();
+
+    /// Watches the specified path.
+    #[MethodInput=CreateWatchInput,rpc_name="file/createWatch",result=create_watch_result,set_result=set_create_watch_result]
+    fn create_watch(&self, path:Path) -> Uuid;
+
+    /// Delete the specified watcher.
+    #[MethodInput=DeleteWatchInput,rpc_name="file/deleteWatch",result=delete_watch_result,
+    set_result=set_delete_watch_result]
+    fn delete_watch(&self, watch_id:Uuid) -> ();
+}
+}
+
+
+
+// =============
+// === Tests ===
+// =============
+
+#[cfg(test)]
+mod tests {
+    use super::*;
+    use super::FileKind::RegularFile;
+
+    use json_rpc::messages::Message;
+    use json_rpc::messages::RequestMessage;
+    use json_rpc::test_util::transport::mock::MockTransport;
+    use serde_json::json;
+    use serde_json::Value;
+    use std::future::Future;
+    use utils::test::poll_future_output;
+    use utils::test::poll_stream_output;
+    use futures::task::LocalSpawnExt;
+
+    struct Fixture {
```

Comment:
What does name `Fixture` mean? I know these are tests but they should be understandable. Either add docs to explain what it is for or change name to more obvious,.

Reply:
`Fixture` is a really common term in unit testing frameworks.

Reply:
Oh, interesting. Havent heard about it before tbh. But if so, lets leave it as is.